dart-impl/mpi/src/dart_team_group.c

```diff
@@ -480,9 +485,13 @@ dart_ret_t dart_group_locality_split(
       for (int d = group_first_dom_idx; d < group_last_dom_idx; ++d) {
         group_num_units += domains[d]->num_units;
       }
-      dart_global_unit_t * group_team_unit_ids =
-                              malloc(sizeof(dart_global_unit_t) *
-                                     group_num_units);
+
+      dart_global_unit_t * group_team_unit_ids = NULL;
+      if(group_num_units != 0){
```

Comment:
This is correct, but if and only if using an up-to-date unix which does not suffer from CERT MEM04-C and CWE-131:

`Undefined allocation of 0 bytes (CERT MEM04-C; CWE-131)`